### PR TITLE
Use the codeQL.runningTests.numberOfThreads

### LINKED ISSUE #625: Setting for specifying threads while running tests is unused
This setting is unused: `codeQL.runningTests.numberOfThreads`.  It is hard-coded to use 8 threads in the code.

### PR DESCRIPTION
**Pull request by aeisenberg**
This setting has existed for a while, but it was not used for some
reason.

Resolves #625.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 - Allow setting `codeQL.runningQueries.numberOfThreads` and `codeQL.runningTests.numberOfThreads` to 0, (which is interpreted as 'use one thread per core on the machine').
 - Clear the problems view of all CodeQL query results when a database is removed.
 - Add a `View DIL` command on query history items. This opens a text editor containing the Datalog Intermediary Language representation of the compiled query.
+- Fix using the codeQL.runningTests.numberOfThreads setting.
```
Comment (adityasharad) [inline extensions/ql-vscode/CHANGELOG.md:16]:
```suggestion
- The `codeQL.runningTests.numberOfThreads` setting is now used correctly when running tests.
```

**Review by adityasharad — APPROVED** [review]
Thank you for spotting this!

File: extensions/ql-vscode/CHANGELOG.md (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/CHANGELOG.md]: Done.

**Review by adityasharad — APPROVED** [review]

**Comment by aeisenberg** [comment]
Checks are not running right now. I will wait until things get fixed.

**Comment by aeisenberg** [comment]
The test failures are legitimate. I fixed them, but I want to add some more tests to be sure.

File: .vscode/launch.json [hunk 2]
```diff
@@ -43,7 +43,6 @@
       "port": 9229,
       "stopOnEntry": false,
       "sourceMaps": true,
-      "preLaunchTask": "Build",
```
Comment (adityasharad) [inline .vscode/launch.json:46]: Deliberate?

File: extensions/ql-vscode/src/config.ts [hunk 3]
```diff
@@ -211,6 +222,21 @@ export class QueryHistoryConfigListener extends ConfigListener implements QueryH
   }
 }
 
+export class CliConfigListener extends ConfigListener implements CliConfig {
+
+  public get numberTestThreads(): number {
+    return NUMBER_OF_TEST_THREADS_SETTING.getValue();
+  }
+
+  public get onDidChangeConfiguration(): Event<void> {
```
Comment (adityasharad) [inline extensions/ql-vscode/src/config.ts:231]: This suggests we could push this common listener property into the superclass. But we can do that later.

**Review by adityasharad — APPROVED** [review]
Glad to have more test coverage in this area!

File: .vscode/launch.json (hunk 2 above)
Comment (aeisenberg) [reply .vscode/launch.json:46]: Yes...that should have been removed a while ago.

Comment (aeisenberg) [reply .vscode/launch.json:46]: It didn't seem useful to create a separate PR for it.

File: extensions/ql-vscode/src/config.ts (hunk 3 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/config.ts:231]: I can do that now.

**Pull request merged**